### PR TITLE
[one-cmds] Revise one-codegen as python script

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit revises one-codegen as python script.

Related : #4278
Draft: #4758
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/one-cmds/one-codegen [hunk 1]
```diff
@@ -14,53 +19,93 @@
...
+    backends_list = []
+    for cand in files:
+        if cand.endswith('-compile'):
+            # 8 : length of '-compile'
+            backends_list.append(cand[:-8])
+    return backends_list
+
+
+def _get_parser():
+    parser = argparse.ArgumentParser(description='command line tool for code generation')
```
Comment (seanshpark) [inline compiler/one-cmds/one-codegen:43]: `command` --> `Command` if there is no other message in front of this.
Please check with all other PRs.

Comment (seanshpark) [reply compiler/one-cmds/one-codegen:43]: confirmed in another PR :)

File: compiler/one-cmds/one-codegen [hunk 2]
```diff
@@ -1,4 +1,9 @@
-#!/bin/bash
+#!/usr/bin/env bash
+''''export SCRIPT_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"                  # '''
+''''export PY_PATH=${SCRIPT_PATH}/venv/bin/python                                       # '''
+''''test -f ${PY_PATH} && exec ${PY_PATH} "$0" "$@"                                     # '''
+''''echo "Error: Virtual environment not found. Please run 'one-prepare-venv' command." # '''
+''''exit 255                                                                            # '''
```
Comment (parjong) [inline compiler/one-cmds/one-codegen:6]: @_@???

File: compiler/one-cmds/one-codegen [hunk 3]
```diff
@@ -1,4 +1,9 @@
-#!/bin/bash
+#!/usr/bin/env bash
+''''export SCRIPT_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"                  # '''
+''''export PY_PATH=${SCRIPT_PATH}/venv/bin/python                                       # '''
+''''test -f ${PY_PATH} && exec ${PY_PATH} "$0" "$@"                                     # '''
```
Comment (parjong) [inline compiler/one-cmds/one-codegen:4]: Ah... this line is the key..

Comment (parjong) [reply compiler/one-cmds/one-codegen:4]: What does ``''''`` mean in BASH?

Comment (parjong) [reply compiler/one-cmds/one-codegen:4]: I guess that it is a string in Python.

Comment (mhs4670go) [reply compiler/one-cmds/one-codegen:4]: Right. `''''    #'''` becomes string in python. ' ' ' ' means `''` + `''`, two string without contents. Another example is like..
```bash
$ 'l''s'
$ "l""s"
```
and so on.

**Review by parjong — APPROVED** [review]
;-)

**Pull request merged**